packages/enzyme-adapter-react-16/src/ReactSixteenAdapter.js

```diff
@@ -426,22 +437,17 @@ class ReactSixteenAdapter extends EnzymeAdapter {
 
           if (!isStateful && isMemo(el.type)) {
             const InnerComp = el.type.type;
-            const wrappedMemoEl = Object.assign(
-              (...args) => InnerComp(...args), // eslint-disable-line new-cap
-              InnerComp,
-            );
-            return withSetStateAllowed(() => renderer.render({ ...el, type: wrappedMemoEl }, context));
+            return withSetStateAllowed(() => renderer.render(
+              { ...el, type: wrapFunctionalComponent(InnerComp) },
+              context,
+            ));
           }
 
           if (!isStateful && typeof Component === 'function') {
-            if (Component !== cachedComponent) {
-              wrappedEl = Object.assign(
-                (...args) => Component(...args), // eslint-disable-line new-cap
-                Component,
-              );
-              cachedComponent = Component;
```

Comment:
this line is important here; we don't want to create a new wrapper every time. Can we preserve it?

Reply:
PR updated with the requested change. I can also simplify the wrap-fuction using lodash.memoize, if you think is worth adding the dep.

Also, a test started failing, I've tracked to this change:
https://github.com/airbnb/enzyme/pull/2014/commits/abed5df0d69afa9eee66cd8ac719f4a213b9b58c (it started failing because it no longer tries to wrap the component)
I've reverted it since React.Memo also work on class component, but let me know if there is a better approach.

Reply:
Nah, no need to add the dep.

Your change seems good :-)